### PR TITLE
Update docs release script so we can have autodeploys

### PR DESCRIPTION
**Pull request by jessfraz**
to docs.master.dockerproject.com.
- Make the invaidation profile the bucket variable, not hard coded.
- Add no cache variable for settings cache to "no-cache"

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle jess@docker.com (github: jfrazelle)

### CODE REVIEW

**Comment by jfrazelle** [comment]
@SvenDowideit @tianon I want/need this now for autodeploy of docs :D

**Comment by tianon** [comment]
Ugg LGTM lol

**Comment by jfrazelle** [comment]
omg really on the first try :D I was so careful w tabs :tada:

**Comment by jfrazelle** [comment]
i am going to bask in this glory all day

File: Makefile [hunk 1]
```diff
@@ -83,7 +83,7 @@ build: bundles
 	docker build -t "$(DOCKER_IMAGE)" .
 
 docs-build:
-	git diff --name-status upstream/release..upstream/docs docs/ > docs/changed-files
+	( git remote | grep -v upstream ) || git diff --name-status upstream/release..upstream/docs docs/ > docs/changed-files
```
Comment (SvenDowideit) [inline Makefile]: mmm, I was thinking to change this to detecting what the user had aliased the `github.com/docker/docker` repo to

**Comment by SvenDowideit** [comment]
meh. LGTM, or you can make it even more awesome :D

**Comment by jfrazelle** [comment]
hmmm hard to do that in one line on the makefile but I was planning on a cleanup PR after this :) i saw a few other thing hehe

**Comment by SvenDowideit** [comment]
sweet :)

File: docs/release.sh [hunk 2]
```diff
@@ -78,6 +78,11 @@ upload_current_documentation() {
 	src=site/
 	dst=s3://$BUCKET$1
 
+	cache="max-age=3600"
+	if [ ! -z "$NOCACHE" ]; then
```
Comment (tianon) [inline docs/release.sh]: Wait, what's this `[ ! -z ... ]` nonsense?  Why not just `[ "$NOCACHE" ]`?

Comment (jfrazelle) [inline docs/release.sh]: dammmmiiitttttt I was so close to being perfect

Comment (tianon) [inline docs/release.sh]: Also, you have some unnecessary interpolation in this same file's changes that I'll leave as an exercise for you to figure out where is at. :wink:

File: docs/release.sh [hunk 3]
```diff
@@ -158,4 +163,4 @@ sed -i "s/^site_url:.*/site_url: \/$MAJOR_MINOR\//" mkdocs.yml
 echo "Building the /$MAJOR_MINOR/ documentation"
 build_current_documentation
 upload_current_documentation "/$MAJOR_MINOR/"
-invalidate_cache "/$MAJOR_MINOR"
+[ "$NOCACHE" ] || invalidate_cache "/$MAJOR_MINOR"
```
Comment (jfrazelle) [inline docs/release.sh]: it's funny bc i did that here and forgot to fix the other, next time im shooting for 100% awesomeness

File: docs/release.sh (hunk 2 above)
Comment (jfrazelle) [inline docs/release.sh]: oooooo

Comment (jfrazelle) [inline docs/release.sh]: i don't see it :cold_sweat:

Comment (tianon) [inline docs/release.sh]: cache="max-age=3600"

and

cache="no-cache"

**Comment by jfrazelle** [comment]
is anything missing for this... I need this so my auto deploy script stops failing :P

File: docs/release.sh [hunk 4]
```diff
@@ -78,6 +78,11 @@ upload_current_documentation() {
 	src=site/
 	dst=s3://$BUCKET$1
 
+	cache="max-age=3600"
+	if [ "$NOCACHE" ]; then
+		cache="no-cache"
```
Comment (tianon) [inline docs/release.sh]: If you want me to review with my super pedantic hat on, this (and the one two lines above) are the ones I was referring to where double quotes aren't actually necessary. :+1:

Comment (tianon) [inline docs/release.sh]: (and if this were in `project/` or `contrib/`, these would definitely need fixing :wink:)

Comment (jfrazelle) [inline docs/release.sh]: it actually needs them tho in the shelling out later, I tried without and it doesn't set properly in aws script them :D

Comment (jfrazelle) [inline docs/release.sh]: let me try again tho

Comment (jfrazelle) [inline docs/release.sh]: nm that works, im going crazy, I was like but it should work.... ok updated lol

**Comment by tianon** [comment]
LGTM
